internal/reader/rmn_home_test.go

```diff
@@ -54,79 +54,62 @@ func TestRMNHomeChainConfigPoller_Ready(t *testing.T) {
 }
 
 func TestRMNHomeChainConfigPoller_HealthReport(t *testing.T) {
-	homeChainReader := readermock.NewMockContractReaderFacade(t)
+	t.Parallel()
 
-	var (
-		tickTime       = 1 * time.Millisecond
-		totalSleepTime = 50 * time.Millisecond // give more time for multiple ticks
-	)
+	tests := []struct {
+		name        string
+		failedPolls uint
+		wantErr     bool
+	}{
+		{
+			name:        "Healthy state",
+			failedPolls: 0,
+			wantErr:     false,
+		},
+		{
+			name:        "Unhealthy state",
+			failedPolls: MaxFailedPolls,
+			wantErr:     true,
+		},
+	}
 
-	// Set up the mock to return an error for the first 10 calls
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Return(fmt.Errorf("error")).Times(50)
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
 
-	// Set up the mock for a successful call after 10 errors
-	primaryConfig, secondaryConfig := createTestRMNHomeConfigs(false, false)
-	rmnHomeOnChainConfigs := []VersionedConfigWithDigest{primaryConfig, secondaryConfig}
+			// Create a mock ContractReaderFacade
+			homeChainReader := readermock.NewMockContractReaderFacade(t)
 
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Run(func(args mock.Arguments) {
-		arg := args.Get(4).(*[]VersionedConfigWithDigest)
-		*arg = rmnHomeOnChainConfigs
-	}).Return(nil).Times(50)
+			// Create the rmnHomePoller
+			poller := NewRMNHomePoller(
+				homeChainReader,
+				rmnHomeBoundContract,
+				logger.Test(t),
+				time.Second,
+			).(*rmnHomePoller)
 
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Return(fmt.Errorf("error"))
+			err := poller.Start(context.Background())
+			require.NoError(t, err)
 
-	configPoller := NewRMNHomePoller(
-		homeChainReader,
-		rmnHomeBoundContract,
-		logger.Test(t),
-		tickTime,
-	)
+			// Set the failedPolls
+			poller.failedPolls = tt.failedPolls
 
-	require.NoError(t, configPoller.Start(context.Background()))
-	// Initially it's healthy
-	healthy := configPoller.HealthReport()
-	require.Equal(t, map[string]error{configPoller.Name(): error(nil)}, healthy)
-
-	// give some time for polling to happen
-	time.Sleep(totalSleepTime)
-	errors := configPoller.HealthReport()
-	require.Equal(t, 1, len(errors))
-	require.Errorf(t, errors[configPoller.Name()], "polling failed %d times in a row", MaxFailedPolls)
-
-	// give some time for successful polling to happen
-	time.Sleep(totalSleepTime * 1)
```

Comment:
General observation, we should not use sleep directly unless necessary. IMO `require.Eventually` is the right way of verifying the condition to be met in some amount of time